src/backend/src/lib.rs

```diff
@@ -154,36 +155,63 @@ pub struct Config {
     pub ecdsa_key_name: String,
     // A list of allowed callers to restrict access to endpoints that do not particularly check or use the caller()
     pub allowed_callers: Vec<Principal>,
+    pub supported_credentials: Option<Vec<SupportedCredential>>,
+    /// Root of trust for checking canister signatures.
+    pub ic_root_key_raw: Option<Vec<u8>>,
```

Comment:
Why do we need this root key in the state? Can't it be derived on the fly when needed? If it has to be saved in the state, what are the security implication?

Reply:
The root key in a local replica changes every time. We already discussed with the crypto team to have a fixed one, but that won't happen in the short term.

Moreover, the root key is not accessible from within a canister.

Therefore, the root key needs to be passed as a parameter on init (or upgrade).

This is how we have been keeping the root key in all the demos we have been doing and the feedback from security was to have a hardcoded mainnet key.

Reply:
Thanks for the exaplantion, did not knew. It feels a bit weird no?
All good.